app/src/main/java/org/wikipedia/views/MessageCardView.kt

```diff
@@ -4,17 +4,43 @@ import android.content.Context
 import android.net.Uri
 import android.util.AttributeSet
 import android.view.View
-import kotlinx.android.synthetic.main.view_suggested_edits_disabled_states.view.*
+import kotlinx.android.synthetic.main.view_message_card.view.*
 import org.wikipedia.R
 import org.wikipedia.analytics.LoginFunnel.SOURCE_SUGGESTED_EDITS
 import org.wikipedia.login.LoginActivity
+import org.wikipedia.util.ResourceUtil
 import org.wikipedia.util.StringUtil
 import org.wikipedia.util.UriUtil
 
-internal class SuggestedEditsDisabledStatesView constructor(context: Context, attrs: AttributeSet? = null) : WikiCardView(context, attrs) {
+internal class MessageCardView constructor(context: Context, attrs: AttributeSet? = null) : WikiCardView(context, attrs) {
 
     init {
-        View.inflate(context, R.layout.view_suggested_edits_disabled_states, this)
+        View.inflate(context, R.layout.view_message_card, this)
+    }
+
+    fun setMessageTitle(title: String) {
+        messageTitleView.text = title
+    }
+
+    fun setMessageText(text: String) {
+        messageTextView.text = text
+    }
+
+    fun setImageSource(imageSource: Int) {
+        if (imageSource < 0) {
```

Comment:
It's not very obvious to implicitly show/hide the image based on the resource parameter.  Perhaps we can have a second (boolean) parameter to show/hide the image.

Reply:
You're correct. Done!